syntax = "proto2";

package tutorial;


message DCC {
  required string version = 1;
  required uint32 algorithm = 2;
  required uint32 kid = 3;
  required uint32 not_before  = 4;
  required uint32 not_after  = 5;
  required string iss = 6;
  required string name = 7;
  required string surname = 8;
  required uint32 date_of_birth = 9;
//  required CertificateType cert_type = 10;
  required uint32 disease = 11;


  message Vaccine {
    required uint32 vaccine = 1;
    required uint32 doses_done = 2;
    required uint32 doses_required = 3;
    required uint32 date_vaccine = 4;
  }

  message Test {
    required bool test_result = 1;
    required uint32 test_used = 2;
    required uint32 date_test = 3;
  }

  message Recovery {
    required uint32 date_test = 1;
    required uint32 date_from = 2;
    required uint32 date_until = 3;
  }

  oneof type {
    Vaccine v = 12;
    Test t = 13;
    Recovery r = 14;
  }

  optional bytes signature = 15;
}